src/tests/CxAuthCall.test.ts

```diff
@@ -1,44 +1,110 @@
-import {CxScanConfigCall} from '../main/CxScanConfigCall';
-import {CxAuthCall} from '../main/CxAuthCall';
+import {CxScanConfig} from '../main/CxScanConfig';
+import {CxAuth} from '../main/CxAuth';
 import {CxParamType} from '../main/CxParamType';
-import CxScan from "../main/CxScan";
 import 'babel-polyfill';
+import {CxCommandOutput} from "../main/CxCommandOutput";
 
-let cxScanConfig = new CxScanConfigCall();
+let cxScanConfig = new CxScanConfig();
 cxScanConfig.baseUri = process.env["CX_BASE_URI"];
 cxScanConfig.clientId = process.env["CX_CLIENT_ID"];
 cxScanConfig.clientSecret = process.env["CX_CLIENT_SECRET"];
 if(process.env["PATH_TO_EXECUTABLE"] !== null && process.env["PATH_TO_EXECUTABLE"] !== undefined ) {
     cxScanConfig.pathToExecutable = process.env["PATH_TO_EXECUTABLE"];
 }
 let params = new Map();
-params.set(CxParamType.PROJECT_NAME, "JayJavascriptWrapperTest");
+params.set(CxParamType.PROJECT_NAME, "ASTJavascriptWrapperTest");
 params.set(CxParamType.SCAN_TYPES, "sast");
-params.set(CxParamType.ADDITIONAL_PARAMETERS, "--nowait");
-params.set(CxParamType.SAST_PRESET_NAME, "Checkmarx Default");
+
 params.set(CxParamType.S, ".");
-const auth = new CxAuthCall(cxScanConfig);
+params.set(CxParamType.FILTER, "*.ts");
+const auth = new CxAuth(cxScanConfig);
 
 describe("ScanCreate cases",() => {
-it('ScanCreate Successful case', async () => {
+    it('ScanCreate Successful case wait mode', async () => {
     const data = await auth.scanCreate(params);
-    const cxScanObject: CxScan = JSON.parse(data);
-    expect(cxScanObject.Status).toContain("Queued");
+    const cxCommandOutput: CxCommandOutput =JSON.parse(JSON.stringify(data))
+        if(cxCommandOutput.scanObjectList.length !== 0) {
+            const scanID = cxCommandOutput.scanObjectList.pop().ID
+            const scanObject = await auth.scanShow(scanID);
+            if(scanObject !== null && scanObject !== undefined) {
+                const status = scanObject.scanObjectList.pop().Status
+                expect(status).toMatch(/(Complete|Running|Queued)/i)
+            }
+        }
+        else {
+            expect(cxCommandOutput.exitCode).toEqual(0)
+        }
 })
+    it('ScanCreate Successful case no wait mode', async () => {
+        params.set(CxParamType.ADDITIONAL_PARAMETERS, "--nowait");
+        const data = await auth.scanCreate(params);
+        const cxCommandOutput: CxCommandOutput =JSON.parse(JSON.stringify(data))
+        if(cxCommandOutput.scanObjectList.length !== 0) {
+            const scanID = cxCommandOutput.scanObjectList.pop().ID
+            const scanObject = await auth.scanShow(scanID);
+            if(scanObject !== null && scanObject !== undefined) {
```

Comment:
We can remove this condition, if scanobject is null or undefined then it should fail in the expect condition